OutOfSchool/OutOfSchool.BusinessLogic/Services/Memento/JSONConverter/JSONCreationConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace OutOfSchool.BusinessLogic.Services.Memento.JSONConverter;
+
+public abstract class JsonCreationConverter<T> : JsonConverter
+{
+    public override bool CanWrite
+    {
+        get
+        {
+            return false;
+        }
+    }
+
+    public override bool CanConvert(Type objectType)
+    {
+        return typeof(T).IsAssignableFrom(objectType);
+    }
+
+    public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+    {
+        ArgumentNullException.ThrowIfNull(reader);
+        ArgumentNullException.ThrowIfNull(serializer);
+
+        if (reader.TokenType == JsonToken.Null)
+        {
+            return null;
+        }
+
+        JObject jObject = JObject.Load(reader);
+        T target = Create(objectType, jObject);
+        serializer.Populate(jObject.CreateReader(), target);
+
+        return target;
+    }
+
+    public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
```

Comment:
(Code style) Isn't it better to just not add this method if you don't implement it?

Reply:
WriteJson() is an abstract method of JsonConverter class, so I have to implemented it.

Reply:
Yes, you must implement abstract methods when inheriting abstract classes, but JsonCreationConverter is also abstract, so there is no need to override it (especially with a simple throwing error). JsonCreationConverter simply extends the JsonConverter's abstaction(if I may put it that way)

Reply:
If I don't implement WriteJson() method in JSONCreationConverter class, then I have to implement it in WorkshopConverter class, which is not very good in my opinion. The only task of WorkshopConverter class is to implement Create() method.